qiskit/primitives/base/base_primitive.py

```diff
@@ -73,31 +74,44 @@ def _validate_circuits(
 
     @staticmethod
     def _validate_parameter_values(
-        parameter_values: Sequence[Sequence[float]] | Sequence[float] | float | None,
-        default: Sequence[Sequence[float]] | Sequence[float] | None = None,
+        parameter_values: Sequence[Sequence[float] | Mapping[ParameterExpression, float]]
+        | Sequence[float]
+        | Mapping[ParameterExpression, float]
+        | float
+        | None,
+        parameter_views: Sequence[ParameterView],
     ) -> tuple[tuple[float, ...], ...]:
-        # Allow optional (if default)
+        # Allow optional
         if parameter_values is None:
-            if default is None:
-                raise ValueError("No default `parameter_values`, optional input disallowed.")
-            parameter_values = default
+            parameter_values = [()] * len(parameter_views)
+
+        # Allow single value
+        if _isreal(parameter_values):
+            parameter_values = [[parameter_values]]
+        elif isinstance(parameter_values, (Sequence, np.ndarray)) and not any(
+            isinstance(vector, (Sequence, Mapping, np.ndarray)) for vector in parameter_values
+        ):
+            parameter_values = [parameter_values]
 
         # Support numpy ndarray
         if isinstance(parameter_values, np.ndarray):
             parameter_values = parameter_values.tolist()
         elif isinstance(parameter_values, Sequence):
-            parameter_values = tuple(
+            parameter_values = [
                 vector.tolist() if isinstance(vector, np.ndarray) else vector
                 for vector in parameter_values
-            )
+            ]
 
-        # Allow single value
-        if _isreal(parameter_values):
-            parameter_values = ((parameter_values,),)
-        elif isinstance(parameter_values, Sequence) and not any(
-            isinstance(vector, Sequence) for vector in parameter_values
-        ):
-            parameter_values = (parameter_values,)
+        # Support Mapping
+        if isinstance(parameter_values, Mapping):
+            parameter_values = [[parameter_values[parameter] for parameter in parameter_views[0]]]
+        elif isinstance(parameter_values, Sequence):
+            parameter_values = [
+                [_validate_and_getitem(value, parameter) for parameter in parameter_view]
```

Comment:
Presumably we need a complete mapping for all parameters in the circuit. If not what is the result here? I recall today if the parameters len do not match you some sort of error to that effect. Will that error message be confusing to a user that passed in a Mapping? As it is below if they pass in a Mapping that happens to have any complex values they will end up the type error below which says it needs a sequence. Maybe we can at least remember the user passed a Mapping and raised a more representative error message?